cmd/mesg-daemon/main.go

```diff
@@ -35,12 +38,29 @@ func main() {
 	// init the config of cosmos
 	cosmos.InitConfig()
 
-	ctx := server.NewDefaultContext()
+	ctx := server.NewContext(
+		cfg.DefaultConfig(),
+		log.NewTMJSONLogger(log.NewSyncWriter(os.Stdout)),
+	)
 	cobra.EnableCommandSorting = false
 	rootCmd := &cobra.Command{
-		Use:               version.ServerName,
-		Short:             "Engine Daemon (server)",
-		PersistentPreRunE: server.PersistentPreRunEFn(ctx),
+		Use:   version.ServerName,
+		Short: "Engine Daemon (server)",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := server.PersistentPreRunEFn(ctx)(cmd, args); err != nil {
+				return err
+			}
+			logger, err := tmflags.ParseLogLevel(ctx.Config.LogLevel, ctx.Logger, cfg.DefaultLogLevel())
```

Comment:
If `PersistentPreRunE` overrides the log then we should override it back ;)
```suggestion
			logger, err := tmflags.ParseLogLevel(ctx.Config.LogLevel, log.NewTMJSONLogger(log.NewSyncWriter(os.Stdout)), cfg.DefaultLogLevel())
```

Reply:
Please see https://github.com/mesg-foundation/engine/pull/1832

Reply:
merged